### PR TITLE
Add amp-ad type sharethrough

### PR DESCRIPTION
**Pull request by nuclearghost**
This is an implementation for [ITI]( https://github.com/ampproject/amphtml/issues/2286)

We originally tried to run our native ads not in an iframe in this [PR](https://github.com/ampproject/amphtml/pull/2271), but since amp isn't quite capable of supporting it at this time. We've added Sharethrough to be used through `amp-ad`

### CODE REVIEW

**Comment by nuclearghost** [comment]
I signed it! I'm a member of the company Sharethrough

File: ads/sharethrough.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
```
Comment (dvoytenko) [inline ads/sharethrough.js]: Could you please also add a reference to `builtins/amp-ad.md` file?

File: ads/sharethrough.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import {writeScript, checkData} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function sharethrough(global, data) {
+  checkData(data, ['pkey']);
+  global.pkey = data.pkey;
```
Comment (dvoytenko) [inline ads/sharethrough.js]: If `pkey` is required, could you please assert it via `validateDataExists`? If not required - ignore.

File: ads/sharethrough.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+</amp-ad>
+```
+
+## Configuration
+
+For semantics of configuration, please [contact Sharethrough](mailto:pubsupport@sharethrough.com).
+
+Supported parameters:
+
+- data-pkey
```
Comment (dvoytenko) [inline ads/sharethrough.md]: If required, please document as such. If not - ignore.

**Comment by dvoytenko** [comment]
@nuclearghost Thanks! Just a couple of comments and it'll be good to go.

**Comment by nuclearghost** [comment]
@dvoytenko Thanks for the feedback. I've updated accordingly

**Comment by dvoytenko** [comment]
@nuclearghost Great! This is ready to be merged. Could you please rebase your PR to master? Looks like there are conflicts right now.

**Comment by nuclearghost** [comment]
@dvoytenko Conflicts resolved!

**Comment by dvoytenko** [comment]
Thanks! Merged.

**Pull request merged**